evalml/tuners/tuner.py

```diff
@@ -89,6 +89,35 @@ def _convert_to_pipeline_parameters(self, flat_parameters):
             pipeline_parameters[component_name][parameter_name] = parameter_value
         return pipeline_parameters
 
+    def get_starting_parameters(self, hyperparameter_ranges):
```

Comment:
This is the main change. Rather than using the hyperparameter ranges attached to the tuner, we can just pass in the hyperparameter ranges that the user specified

Reply:
I think this fixes the problem with iterative algorithm but not the default algorithm. When we go to the "fast final" batch, we'll take some parameters from the tuner and not use the default ones.

This is the logic now for creating the pipelines in the fast final batch, note that we just call `propose`. 
```python
    def _create_n_pipelines(self, pipelines, n):
        next_batch = []
        for _ in range(n):
            for pipeline in pipelines:
                if pipeline.name not in self._tuners:
                    self._create_tuner(pipeline)

                select_parameters = self._create_select_parameters()
                proposed_parameters = self._tuners[pipeline.name].propose()
                parameters = self._transform_parameters(pipeline, proposed_parameters)
                parameters.update(select_parameters)
                next_batch.append(
                    pipeline.new(parameters=parameters, random_seed=self.random_seed)
                )
        return next_batch
```

In main, we don't take a sample from the tuner:
```python
    def _create_fast_final(self):
        estimators = [
            estimator
            for estimator in get_estimators(self.problem_type)
            if estimator not in self._naive_estimators()
        ]
        estimators = self._filter_estimators(
            estimators,
            self.problem_type,
            self.allow_long_running_models,
            None,
            self.y.nunique(),
            self.logger,
        )
        pipelines = self._make_pipelines_helper(estimators)

        if self._split:
            self._rename_pipeline_parameters_custom_hyperparameters(pipelines)

        next_batch = []
        for pipeline in pipelines:
            parameters = self._create_select_parameters()
            pipeline = pipeline.new(
                parameters=self._transform_parameters(pipeline, parameters),
                random_seed=self.random_seed,
            )
            next_batch.append(pipeline)

        for pipeline in next_batch:
            self._create_tuner(pipeline)
        return next_batch
```

And then in `_transform_parameters` we make sure we we take a sample if custom hyperparameters are present - 
```python
    def _create_fast_final(self):
        estimators = [
            estimator
            for estimator in get_estimators(self.problem_type)
            if estimator not in self._naive_estimators()
        ]
        estimators = self._filter_estimators(
            estimators,
            self.problem_type,
            self.allow_long_running_models,
            None,
            self.y.nunique(),
            self.logger,
        )
        pipelines = self._make_pipelines_helper(estimators)

        if self._split:
            self._rename_pipeline_parameters_custom_hyperparameters(pipelines)

        next_batch = []
        for pipeline in pipelines:
            parameters = self._create_select_parameters()
            pipeline = pipeline.new(
                parameters=self._transform_parameters(pipeline, parameters),
                random_seed=self.random_seed,
            )
            next_batch.append(pipeline)

        for pipeline in next_batch:
            self._create_tuner(pipeline)
        return next_batch
```

On this branch, nyc taxi takes 10 minutes on the default algoirhtm
![image](https://user-images.githubusercontent.com/41651716/161077336-e59e33fe-98d2-427a-a7c4-8b397750c65c.png)

But one minute on main:
![image](https://user-images.githubusercontent.com/41651716/161077445-07e39829-123e-4e20-a5b4-82be4e195681.png)

Reply:
Great catch @freddyaboulton! I will fix this and run tests locally to make sure it works as expected.